Fix Nuget Update command to respect repositoryPath in nuget.config

slovely opened the pull request:
Executing the following command line:

```
nuget update mysolution.sln
```

will update all the packages in a solution.  However, if there is custom `repositoryPath` specified in a nuget.config, it will error with `Unable to locate the packages folder. Try specifying it using the repositoryPath switch`.  Updating a single package (with a project file) works as expected.

This Pull Request is a simple update to ensure that a custom `packages` folder is respected.  It will fallback to assuming `SolutionDir+"\packages"` if no config file is present.

slovely committed:
Update Nuget Update command to load repositoryPath from nuget.config file (if present).

dnfclas commented:
Hi **@slovely**, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no Contribution License Agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, DNFBOT;

yishaigalatzer commented:
The fix looks good, please add a test to verify all the cases (relative and absolute paths)

yishaigalatzer commented:
@slovely ping, we plan to close the 3.3 release today, so if you can add tests today we might be able to get it into this release

slovely commented:
Thanks @yishaigalatzer - I have had difficulty getting the `build-nuget.ps1` script running correctly and haven't had the time to spend figuring it out yet (my first dabble with dnx).  Will miss the 3.3 release I'm afraid - getting it working and adding tests is on my list though.

Sorry that's not a very helpful response.

yishaigalatzer commented:
No worries, we appreciate what you are doing, and it will make it into the following release.

The easy things that might make it work for you are:
1. You want to use DNX Beta7 (not anything later) at this point, though we will probably update to RC-1 shortly.
2. You want to run build.ps1 with CleanCache flag, and also git clean -xdf the first time.

slovely commented:
Apologies for the time taken to update this PR.  Added tests for custom `repositoryPath` settings in nuget.config when updating entire solution.

I rebased and force-pushed over my original commit - hope that is ok!

Merry Christmas :christmas_tree: :santa:

yishaigalatzer commented:
@alpaix can you please review (add fixes on top if needed) and merge?

alpaix commented:
Looks good! Merging...

@slovely Thanks for the contribution!
